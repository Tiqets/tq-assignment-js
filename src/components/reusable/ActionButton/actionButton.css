.actionButton {
  height: 50px;
  margin: 0px 10px;
  background-color: #018794;
  color: white;
  font-size: 16px;
  border: 0px;
  border-radius: 5px;
  padding: 0px 30px;
}