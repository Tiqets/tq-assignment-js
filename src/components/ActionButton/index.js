import React from 'react';
import PropTypes from 'prop-types';
import './actionButton.css';

/* Feel free to make this a functional component if you consider it necessary */
const ActionButton = (props) => {
  const { label, role } = props;

  return (
    <button className="actionButton" type="button" role={role}>{label}</button>
  );
};

ActionButton.propTypes = {
  label: PropTypes.string,
  role: PropTypes.string,
};

ActionButton.defaultProps = {
  label: 'Search',
  role: 'button',
};

export default ActionButton;
