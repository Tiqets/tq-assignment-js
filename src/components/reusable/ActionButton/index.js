import React from 'react';
import PropTypes from 'prop-types';
import './actionButton.css';

/* Feel free to make this a functional component if you consider it necessary */
const ActionButton = (props) => {
  const { label, role, onSubmit } = props;

  return (
    <button className="actionButton" type="button" role={role} onClick={onSubmit}>{label}</button>
  );
};

ActionButton.propTypes = {
  label: PropTypes.string,
  role: PropTypes.string,
  onSubmit: PropTypes.func,
};

ActionButton.defaultProps = {
  label: 'Search',
  role: 'button',
  onSubmit: () => false,
};

export default ActionButton;
