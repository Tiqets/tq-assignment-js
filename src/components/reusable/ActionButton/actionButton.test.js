/* global it, expect, document */
import React from 'react';
import ReactDOM from 'react-dom';
import TestRenderer from 'react-test-renderer';
import ActionButton from './index';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<ActionButton />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('Handles the label prop correctly', () => {
  const div = document.createElement('div');
  ReactDOM.render(<ActionButton label="new label" />, div);
  expect(div.textContent).toBe('new label');
});

it('renders the action button correctly with default props', () => {
  const TextInputComponent = TestRenderer.create(<ActionButton />).toJSON();
  expect(TextInputComponent).toMatchSnapshot();
});
