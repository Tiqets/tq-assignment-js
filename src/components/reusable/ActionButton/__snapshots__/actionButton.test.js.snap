// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the action button correctly with default props 1`] = `
<button
  className="actionButton"
  onClick={[Function]}
  role="button"
  type="button"
>
  Search
</button>
`;
